libs/providers/flipt-web/src/lib/flipt-web-provider.ts

```diff
@@ -0,0 +1,263 @@
+      .then(() => {
+        this._globalContext = context || {};
+        this._status = ProviderStatus.READY;
+        this._logger?.info('FliptWebProvider initialized');
+      })
+      .catch((error) => {
+        this._status = ProviderStatus.ERROR;
+        this._logger?.error('FliptWebProvider initialization failed', error);
+      });
+  }
+
+  async initializeClient() {
+    let authentication = {};
+
+    if (this._options?.authentication && 'clientToken' in this._options.authentication) {
+      authentication = { client_token: this._options.authentication.clientToken };
+    } else if (this._options?.authentication && 'jwtToken' in this._options.authentication) {
+      authentication = { jwt_token: this._options.authentication.jwtToken };
+    }
+
+    try {
+      this._client = await FliptEvaluationClient.init(this._namespace || 'default', {
+        url: this._options?.url || 'http://localhost:8080',
+        fetcher: this._options?.fetcher,
+        authentication,
+      });
+    } catch (e) {
+      throw new GeneralError(getErrorMessage(e));
+    }
+  }
+
+  async onContextChange(_oldContext: EvaluationContext, newContext: EvaluationContext): Promise<void> {
+    this._globalContext = newContext;
+    await this._client?.refresh();
+  }
+
+  resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context: EvaluationContext,
+  ): ResolutionDetails<boolean> {
+    const mergedContext = { ...this._globalContext, ...context };
+    const evalContext: Record<string, string> = transformContext(mergedContext);
+
+    try {
+      const resp = this._client?.evaluateBoolean(flagKey, mergedContext.targetingKey ?? '', evalContext);
+
+      if (resp?.status === 'failure') {
+        return {
+          value: defaultValue,
+          errorCode: ErrorCode.GENERAL,
+          errorMessage: resp.error_message,
+          reason: StandardResolutionReasons.ERROR,
+        };
+      }
+
+      const result = resp?.result;
+
+      switch (result?.reason) {
+        case EvaluationReason.DEFAULT:
+          return {
+            value: result?.enabled,
+            reason: StandardResolutionReasons.TARGETING_MATCH,
+          };
+        case EvaluationReason.MATCH:
+          return {
+            value: result?.enabled,
+            reason: StandardResolutionReasons.TARGETING_MATCH,
+          };
+        case EvaluationReason.UNKNOWN:
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.UNKNOWN,
+          };
+        default:
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.DEFAULT,
+          };
+      }
+    } catch (e) {
+      throw new GeneralError(getErrorMessage(e));
+    }
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    context: EvaluationContext,
+  ): ResolutionDetails<string> {
+    const mergedContext = { ...this._globalContext, ...context };
```

Comment:
This shouldn't be necessary. The context should be merged by the SDK before the resolve method is called.

Reply:
so does that mean we dont need `_globalContext` at all?